### ISSUE TITLE
Remove force auto-applying global scopes

### ISSUE THREAD

**Issue by antonkomarev** [cybercog/laravel-eloquent-flag#34]
After start using this package in different projects I've understood that has made wrong decision at the start.

I've started to build classes with a lot of such methods to don't use auto apply and have more hand control:
```php
public function shouldApplyPublishedAtScope()
{
    return false;
}
```

I'm thinking to invert this behavior in next major version.

*Maybe it is a good starting point to make configuration file for this package.*

**Comment by lon9man** [comment]
i thought about potential use case:
user has boolean setting (which we put in session) saying **include_archive** and ONLY when it equals to **true** then we should apply global query scope **archieved_at**.

is this case similar to your thoughts in the first post?

**Comment by lon9man** [comment]
do you plan to release new major version? maybe some roadmap with changes and dates?

**Comment by antonkomarev** [comment]
Definitely this change requires to release major version. This is the only one thing I want to change so far, so change log will be tiny. But it will require some changes in client applications. I'll try to find some free time before the New Year.

**Comment by lon9man** [comment]
Счастливого Нового года так сказать)

**Comment by antonkomarev** [comment]
Thanks. I've reviewed a code again and found that some traits are missing this feature completely and some methods names looks strange for me now. Let's place it on hold and take more time for a refactoring process.

**Comment by lon9man** [comment]
how i can subscribe for news related to your changes?

**Comment by antonkomarev** [comment]
You can [Watch repository](https://help.github.com/articles/watching-and-unwatching-repositories/#watching-a-single-repository).
And because you've replied in this issue, you was added as participant to it, what means you will receive emails for every new comment here.

**Comment by antonkomarev** [comment]
All required changes were done in PR #56. Will be included in v5.0
